qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -46,15 +51,20 @@ def __init__(
         basis_gates=None,
         approximation_degree=1.0,
         target=None,
+        decomposer=None,
     ):
         """ConsolidateBlocks initializer.
 
+        The decomoposer used is determined by the first of the following arguments
+        with a non-None value: decomposer, kak_basis_gate, basis_gates. If all are None,
+        then a default decomposer is used.
         Args:
             kak_basis_gate (Gate): Basis gate for KAK decomposition.
-            force_consolidate (bool): Force block consolidation
+            force_consolidate (bool): Force block consolidation.
             basis_gates (List(str)): Basis gates from which to choose a KAK gate.
             approximation_degree (float): a float between [0.0, 1.0]. Lower approximates more.
-            target (Target): The target object for the compilation target backend
+            target (Target): The target object for the compilation target backend.
+            decomposer: A 2q gate decomposer.
```

Comment:
I think we should have a bit more detail on what this argument is or mark it as private. As a user I'm not clear what the input type needs to be here or what the constraints are on the input type. 

That being said, I'm not clear on why this argument is needed. I can see the argument for making this an input longer term so that users with custom decomposers could use that here. But in actuality the decomposer isn't really used for much in this pass, it just checks the number of supported qubits and the two qubit gate used in the kak decomposition (so as to not resynthesize a block unnecessarily).

Reply:
The argument `decomposer`  was introduced just for supporting control flow ops.  I added it to avoid constructing a new decomposer. I don't know of any other use for it. I agree it makes sense not to expose it. We could construct a new decomposer using `__new__`. That would be inelegant, but would avoid introducing the new argument.

But if we can reuse the same instance of `ConsolidateBlocks`, then the whole question is moot.

Reply:
This argument was removed in 3da4990

Reply:
I also removed any mention of the decomposer in bf0a1f4.  btw. the decomposer is needed for
https://github.com/Qiskit/qiskit-terra/blob/e2674cebcc9cf2cd7333f7fe645a833be27bc972/qiskit/quantum_info/synthesis/two_qubit_decompose.py#L1409-L1411